newsletter/models.py

```diff
@@ -584,6 +585,7 @@ def submit(self):
         assert self.publish_date < now(), \
             'Something smells fishy; submission time in future.'
 
+        signals.pre_submit.send(sender=self.__class__, message=self)
```

Comment:
The parameter shouldn't be called message if the object passed isn't a message but the submission object.

```suggestion
        signals.pre_submit.send(sender=self.__class__, submission=self)
```